components/support/nimbus-fml/src/util/loaders.rs

```diff
@@ -120,15 +170,64 @@ impl FileLoader {
         Ok(Self {
             cache_dir,
             fetch_client: http_client,
+            cwd,
 
             config,
         })
     }
 
+    #[cfg(test)]
     pub fn default() -> Result<Self> {
         let cwd = std::env::current_dir()?;
         let cache_path = cwd.join("build/app/fml-cache");
-        Self::new(cache_path, Default::default())
+        Self::new(
+            cache_path,
+            std::env::current_dir().expect("CWD not set"),
+            Default::default(),
+        )
+    }
+
+    pub fn add_repo_file(&mut self, file: &FilePath) -> Result<()> {
+        let string = self.read_to_string(file)?;
+
+        let config: BTreeMap<String, String> = if file.to_string().ends_with(".json") {
+            serde_json::from_str(&string)?
+        } else {
+            serde_yaml::from_str(&string)?
+        };
+
+        for (k, v) in config {
+            self.add_repo(file, k, v)?;
+        }
+
+        Ok(())
+    }
+
+    fn add_repo(&mut self, cwd: &FilePath, k: String, v: String) -> Result<(), FMLError> {
+        let k = k.replacen('@', "", 1);
+        let v = if v.ends_with('/') {
+            v
+        } else {
+            format!("{}/", v)
+        };
+        let v =
+            if v.starts_with('.') || v.starts_with('/') || v.contains(":\\") || v.contains("://") {
+                cwd.join(&v)?
+            } else {
+                self.remote_file_path(&k, &v)?
+            };
+        self.config.insert(k, v);
+        Ok(())
+    }
+
+    fn remote_file_path(&self, repo: &str, branch_or_tag: &str) -> Result<FilePath, FMLError> {
+        let base_url = format!("{}/{}/{}", GITHUB_USER_CONTENT_DOTCOM, repo, branch_or_tag);
```

Comment:
I almost want to be even more strict and force a specific git commit/revision, to help make sure that the files don't shift from underneath a developer when they do a full build